temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -217,21 +217,21 @@ public WorkflowStateMachines(
    *     and triggered an execution. Used in {@link WorkflowTaskStateMachine} only to understand
    *     that this workflow task will not have a matching closing event and needs to be executed.
    */
-  public void setWorklfowStartedEventId(long workflowTaskStartedEventId) {
+  public void setWorkflowStartedEventId(long workflowTaskStartedEventId) {
     this.workflowTaskStartedEventId = workflowTaskStartedEventId;
   }
 
-  public void setLastWFTStartedEventId(long eventId) {
+  public void resetStartedEvenId(long eventId) {
+    // When we reset the event ID on a speculative WFT we need to move this counter back
+    // to account for the speculative WFT.
+    long resetLastHandledEventId = this.lastHandledEventId - 2;
```

Comment:
Best to say why it's `- 2`

Reply:
That is what the comment  above is meant to explain

Reply:
Right, it says move back but not why by two. Equivalent comment in core: https://github.com/temporalio/sdk-core/blob/90db5b485f8805902834f48c5ed87f6de5cdaac9/core/src/worker/workflow/machines/workflow_machines.rs#L477

Reply:
I'll probably just plagiarize this comment :)